app/controllers/admin/facility_groups_controller.rb

```diff
@@ -89,24 +96,4 @@ def facility_group_params
       remove_block_ids: []
     )
   end
-
-  def update_params
-    facility_group_params.except(:state)
-  end
-
-  def create_successful?
-    @facility_group.create_state_region!
-
-    if @facility_group.save && @facility_group.toggle_diabetes_management
-      @facility_group.update_block_regions!
```

Comment:
The above three operations were all operating in their own separate transaction, creating a lot of places where things can fail and we are left with a FacilityGroup and/or Region in an invalid state that is out of sync. This PR is refactoring towards a better pattern where we can better see all the things going on w/ create and update, so that we can keep everything in a single transaction for data integrity.

Reply:
Isn't the whole action wrapped in a single transaction and all 3-4 things happen together? We can move that inside the function if it reads better, I think. But logically, it seems like it's doing the right thing.

Reply:
Yup, but wrapping the entire action is too broad. We really want the transaction only around the database level concerns, and the controller action will be doing other things beyond just saving the models (it could kick off jobs, or send an email, or render the view...etc).

I don't know the behavior if a transaction were to fail when the view has already started rendering, but I don't want to try and find out :). We just want to minimize the scope of transactions in general to the actual model level changes that need it.

Reply:
That's fair, I think it could just be within `create_successful?` and `update_successful?` – which I guess is what you've effectively done.

Reply:
Yeah, I'd like to have it in the model layer entirely somehow, but that gets awkward with how much the controller is coupled to the model now via virtual attrs.  This is a good enough first step.